qemu/tests/boot_with_max_queues.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_net
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Boot guest with different vectors, then do netperf testing.
+
+    1) Boot up VM with max vectors.
+    2) Check guest msi & queues info
+    3) Start 10 scp file transfer tests
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the tes t parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    login_timeout = int(params.get("login_timeout", 360))
+    cmd_timeout = int(params.get("cmd_timeout", 240))
+
+    # boot the vm with the max queues
+    queues = int(params["queues"])
+    error_context.context("Boot the guest with queues = %s" % queues,
+                          logging.info)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    # enable max queues for linux guest
+    if params["os_type"] == "linux":
+        nic = vm.virtnet[0]
+        ifname = utils_net.get_linux_ifname(session, nic.mac)
+        set_queue_cmd = "ethtool -L %s combined %s" % (ifname, queues)
+        status, output = session.cmd_status_output(set_queue_cmd,
+                                                   timeout=cmd_timeout,
+                                                   safe=True)
+        if status and "combined unmodified, ignoring" not in output:
+            err = "Failed to set queues to %s with status = %s and output= %s"
+            err %= (queues, status, output)
+            test.error(err)
```

Comment:
@kimi1978wy How about using test.fail here? As it's not caused by the env, but the cmd doesn't perform well.

Reply:
@pezhang yes, you're right, modified, pls review it again :)